lib/ui/platform_dispatcher.dart

```diff
@@ -1158,6 +1168,22 @@ class PlatformDispatcher {
     _onSemanticsActionZone = Zone.current;
   }
 
+  /// A callback that is invoked whenever the user requests an action to be
+  /// performed on a semantics node.
+  ///
+  /// This callback is used when the user expresses the action they wish to
+  /// perform based on the semantics node supplied by updateSemantics.
+  ///
+  /// The framework invokes this callback in the same zone in which the
+  /// callback was set.
+  SemanticsActionEventCallback? get onSemanticsActionEvent => _onSemanticsActionEvent;
+  SemanticsActionEventCallback? _onSemanticsActionEvent;
+  Zone _onSemanticsActionEventZone = Zone.root;
+  set onSemanticsActionEvent(SemanticsActionEventCallback? callback) {
```

Comment:
From my understanding, the engine will not invoke the `SemanticsActionEventCallback` currently. Should we make that work as part of this pull request before making this API available to users?

Reply:
Ah sorry, I can't read. I thought that'd be in `hooks.dart`, but your approach is good too. Marking as resolved.